staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -181,7 +181,7 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 				if err := json.Unmarshal(modified, obj); err != nil {
 					return true, nil, err
 				}
-			case types.StrategicMergePatchType:
+			case types.StrategicMergePatchType, types.ApplyPatchType:
```

Comment:
Indeed, it might be problematic in some cases. I've done it to enable adjusting the unit tests. I'm wondering how we should fix it. Some options that come to me:
1. provide a complete implementation in the testing library that would mimic the server (would be preferred, but is it feasible?)
2. detect some unsupported uses in the testing library and return an error, still allowing for simple test cases to work
3. revert the change, but then it isn't clear to me how to write / adjust existing unit tests

Reply:
I would start by reverting the change to avoid misleading random unit tests that try to use apply

To enable specific unit tests, they could add a reactor that handles this patch type, asserts receiving specific apply patch content, and mocks the application of that patch to the test object

Reply:
Ok, I have opened the issue for now: https://github.com/kubernetes/kubernetes/issues/116851. Hope there are some contributors who can help with that.